File: src/main/java/co/aplicared/jvm/juego/arbúcies/control/Mouse.java
package co.aplicared.jvm.juego.arbúcies.control;

import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;
import java.awt.event.MouseMotionListener;


public class Mouse implements MouseListener, MouseMotionListener {

    private static int mouseX = -1, mouseY = -1, mouseB = -1;

    public static int getMouseX() {
        return mouseX;
    }

    public static int getMouseY() {
        return mouseY;
    }

    public static int getMouseB() {
        return mouseB;
    }

    @Override
    public void mouseClicked(MouseEvent e) {

    }

    @Override
    public void mousePressed(MouseEvent e) {
        mouseB = e.getButton();
    }

    @Override
    public void mouseReleased(MouseEvent e) {
        mouseB = -1;
    }

    @Override
    public void mouseEntered(MouseEvent e) {

    }

    @Override
    public void mouseExited(MouseEvent e) {

    }

    @Override
    public void mouseDragged(MouseEvent e) {

    }

    @Override
    public void mouseMoved(MouseEvent e) {
        mouseX = e.getX();
        mouseY = e.getY();
    }
}

File: src/main/kotlin/co/aplicared/jvm/juego/arbucies/control/Mouse.kt
package co.aplicared.jvm.juego.arbucies.control

import java.awt.event.MouseEvent
import java.awt.event.MouseListener
import java.awt.event.MouseMotionListener

class Mouse : MouseListener, MouseMotionListener {

    companion object {
        var mouseX = -1
        var mouseY = -1
        var mouseB = -1
    }

    override fun mouseReleased(e: MouseEvent?) {
        mouseB = -1
    }

    override fun mouseEntered(e: MouseEvent?) {
    }

    override fun mouseClicked(e: MouseEvent?) {
    }

    override fun mouseExited(e: MouseEvent?) {
    }

    override fun mousePressed(e: MouseEvent?) {
        mouseB = e!!.button
    }

    override fun mouseMoved(e: MouseEvent?) {
        mouseX = e!!.x
        mouseY = e.y
    }

    override fun mouseDragged(e: MouseEvent?) {
        mouseX = e!!.x
        mouseY = e.y
    }
}